### ISSUE TITLE
Input function tutorial doesn't deliver reported performance.

### ISSUE THREAD

**Issue by adrinjalali** [tensorflow/tensorflow#12707]
Running the example explained [here](https://www.tensorflow.org/get_started/input_fn), results in a reported loss 100 times bigger than what's reported in the tutotrial. It also doesn't converge, and it seems to be going up and down.

The expected performance is:

```
INFO:tensorflow:Step 1: loss = 483.179
INFO:tensorflow:Step 101: loss = 81.2072
INFO:tensorflow:Step 201: loss = 72.4354
...
INFO:tensorflow:Step 1801: loss = 33.4454
INFO:tensorflow:Step 1901: loss = 32.3397
INFO:tensorflow:Step 2001: loss = 32.0053
INFO:tensorflow:Step 4801: loss = 27.2791
INFO:tensorflow:Step 4901: loss = 27.2251
INFO:tensorflow:Saving checkpoints for 5000 into /tmp/boston_model/model.ckpt.
INFO:tensorflow:Loss for final step: 27.1674.
```

Whereas, here is what I get when I run it:

```
$ python2 boston.py 
INFO:tensorflow:Using default config.
INFO:tensorflow:Using config: {'_save_checkpoints_secs': 600, '_session_config': None, '_keep_checkpoint_max': 5, '_tf_random_seed': 1, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_save_checkpoints_steps': None, '_model_dir': '/tmp/boston_model', '_save_summary_steps': 100}
INFO:tensorflow:Create CheckpointSaverHook.
2017-08-30 16:38:36.378259: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-30 16:38:36.378318: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-30 16:38:36.378321: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-08-30 16:38:36.378325: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-30 16:38:36.378328: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
INFO:tensorflow:Restoring parameters from /tmp/boston_model/model.ckpt-5000
INFO:tensorflow:Saving checkpoints for 5001 into /tmp/boston_model/model.ckpt.
INFO:tensorflow:loss = 3897.4, step = 5001
INFO:tensorflow:global_step/sec: 476.014
INFO:tensorflow:loss = 2826.95, step = 5101 (0.210 sec)
INFO:tensorflow:global_step/sec: 485.788
INFO:tensorflow:loss = 4138.43, step = 5201 (0.206 sec)
INFO:tensorflow:global_step/sec: 475.238
INFO:tensorflow:loss = 3182.46, step = 5301 (0.210 sec)
INFO:tensorflow:global_step/sec: 477.947
INFO:tensorflow:loss = 4233.69, step = 5401 (0.209 sec)
INFO:tensorflow:global_step/sec: 484.752
INFO:tensorflow:loss = 3593.6, step = 5501 (0.206 sec)
INFO:tensorflow:global_step/sec: 487.786
INFO:tensorflow:loss = 3316.72, step = 5601 (0.205 sec)
INFO:tensorflow:global_step/sec: 493.849
INFO:tensorflow:loss = 3041.69, step = 5701 (0.203 sec)
INFO:tensorflow:global_step/sec: 462.593
INFO:tensorflow:loss = 2753.05, step = 5801 (0.216 sec)
INFO:tensorflow:global_step/sec: 468.178
INFO:tensorflow:loss = 4609.25, step = 5901 (0.214 sec)
INFO:tensorflow:global_step/sec: 505.747
INFO:tensorflow:loss = 5840.77, step = 6001 (0.198 sec)
INFO:tensorflow:global_step/sec: 484.518
INFO:tensorflow:loss = 4149.44, step = 6101 (0.206 sec)
INFO:tensorflow:global_step/sec: 516.622
INFO:tensorflow:loss = 2485.26, step = 6201 (0.193 sec)
INFO:tensorflow:global_step/sec: 480.998
INFO:tensorflow:loss = 3849.6, step = 6301 (0.208 sec)
INFO:tensorflow:global_step/sec: 480.866
INFO:tensorflow:loss = 3684.53, step = 6401 (0.208 sec)
INFO:tensorflow:global_step/sec: 468.132
INFO:tensorflow:loss = 2943.32, step = 6501 (0.214 sec)
INFO:tensorflow:global_step/sec: 463.973
INFO:tensorflow:loss = 3135.92, step = 6601 (0.216 sec)
INFO:tensorflow:global_step/sec: 464.991
INFO:tensorflow:loss = 4463.47, step = 6701 (0.215 sec)
INFO:tensorflow:global_step/sec: 386.855
INFO:tensorflow:loss = 3488.36, step = 6801 (0.259 sec)
INFO:tensorflow:global_step/sec: 402.492
INFO:tensorflow:loss = 3472.3, step = 6901 (0.248 sec)
INFO:tensorflow:global_step/sec: 468.325
INFO:tensorflow:loss = 6648.4, step = 7001 (0.213 sec)
INFO:tensorflow:global_step/sec: 482.791
INFO:tensorflow:loss = 5161.27, step = 7101 (0.207 sec)
INFO:tensorflow:global_step/sec: 468.145
INFO:tensorflow:loss = 6147.99, step = 7201 (0.214 sec)
INFO:tensorflow:global_step/sec: 465.968
INFO:tensorflow:loss = 5066.78, step = 7301 (0.215 sec)
INFO:tensorflow:global_step/sec: 464.296
INFO:tensorflow:loss = 3810.97, step = 7401 (0.215 sec)
INFO:tensorflow:global_step/sec: 464.02
INFO:tensorflow:loss = 3235.41, step = 7501 (0.215 sec)
INFO:tensorflow:global_step/sec: 465.755
INFO:tensorflow:loss = 4455.11, step = 7601 (0.215 sec)
INFO:tensorflow:global_step/sec: 445.682
INFO:tensorflow:loss = 2335.67, step = 7701 (0.224 sec)
INFO:tensorflow:global_step/sec: 446.977
INFO:tensorflow:loss = 4264.51, step = 7801 (0.224 sec)
INFO:tensorflow:global_step/sec: 473.768
INFO:tensorflow:loss = 2708.95, step = 7901 (0.211 sec)
INFO:tensorflow:global_step/sec: 470.603
INFO:tensorflow:loss = 4107.13, step = 8001 (0.212 sec)
INFO:tensorflow:global_step/sec: 494.138
INFO:tensorflow:loss = 2911.78, step = 8101 (0.202 sec)
INFO:tensorflow:global_step/sec: 504.535
INFO:tensorflow:loss = 7470.54, step = 8201 (0.198 sec)
INFO:tensorflow:global_step/sec: 516.137
INFO:tensorflow:loss = 2581.56, step = 8301 (0.195 sec)
INFO:tensorflow:global_step/sec: 503.267
INFO:tensorflow:loss = 4270.98, step = 8401 (0.198 sec)
INFO:tensorflow:global_step/sec: 528.762
INFO:tensorflow:loss = 4186.45, step = 8501 (0.189 sec)
INFO:tensorflow:global_step/sec: 509.902
INFO:tensorflow:loss = 5767.94, step = 8601 (0.196 sec)
INFO:tensorflow:global_step/sec: 510.832
INFO:tensorflow:loss = 3960.38, step = 8701 (0.196 sec)
INFO:tensorflow:global_step/sec: 525.036
INFO:tensorflow:loss = 4677.09, step = 8801 (0.190 sec)
INFO:tensorflow:global_step/sec: 522.431
INFO:tensorflow:loss = 1912.38, step = 8901 (0.191 sec)
INFO:tensorflow:global_step/sec: 521.464
INFO:tensorflow:loss = 2646.11, step = 9001 (0.192 sec)
INFO:tensorflow:global_step/sec: 523.779
INFO:tensorflow:loss = 2803.72, step = 9101 (0.191 sec)
INFO:tensorflow:global_step/sec: 517.767
INFO:tensorflow:loss = 2995.35, step = 9201 (0.193 sec)
INFO:tensorflow:global_step/sec: 467.628
INFO:tensorflow:loss = 3767.14, step = 9301 (0.214 sec)
INFO:tensorflow:global_step/sec: 463.383
INFO:tensorflow:loss = 3825.25, step = 9401 (0.215 sec)
INFO:tensorflow:global_step/sec: 554.173
INFO:tensorflow:loss = 3352.59, step = 9501 (0.180 sec)
INFO:tensorflow:global_step/sec: 528.989
INFO:tensorflow:loss = 4678.92, step = 9601 (0.189 sec)
INFO:tensorflow:global_step/sec: 490.677
INFO:tensorflow:loss = 2191.4, step = 9701 (0.204 sec)
INFO:tensorflow:global_step/sec: 463.207
INFO:tensorflow:loss = 5998.51, step = 9801 (0.216 sec)
INFO:tensorflow:global_step/sec: 461.608
INFO:tensorflow:loss = 3440.55, step = 9901 (0.217 sec)
INFO:tensorflow:Saving checkpoints for 10000 into /tmp/boston_model/model.ckpt.
INFO:tensorflow:Loss for final step: 4595.88.
INFO:tensorflow:Starting evaluation at 2017-08-30-14:38:47
INFO:tensorflow:Restoring parameters from /tmp/boston_model/model.ckpt-10000
INFO:tensorflow:Finished evaluation at 2017-08-30-14:38:47
INFO:tensorflow:Saving dict for global step 10000: average_loss = 16.7064, global_step = 10000, loss = 1670.64
Loss: 1670.643555
INFO:tensorflow:Restoring parameters from /tmp/boston_model/model.ckpt-10000
Predictions: [array([ 34.28430557], dtype=float32), array([ 18.48677063], dtype=float32), array([ 24.49580383], dtype=float32), array([ 34.83223343], dtype=float32), array([ 15.90157318], dtype=float32), array([ 20.3986187], dtype=float32)]
```

Environment:

== cat /etc/issue ===============================================
Linux adrin-leni.ancud.de 4.12.7-1-ARCH #1 SMP PREEMPT Sun Aug 13 08:17:09 CEST 2017 x86_64 GNU/Linux

== are we in docker =============================================
No

== compiler =====================================================
c++ (GCC) 7.2.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

== uname -a =====================================================
Linux adrin-leni.ancud.de 4.12.7-1-ARCH #1 SMP PREEMPT Sun Aug 13 08:17:09 CEST 2017 x86_64 GNU/Linux

== check pips ===================================================
numpy (1.13.1)
protobuf (3.4.0)
tensorflow (1.3.0)
tensorflow-tensorboard (0.1.5)

== check for virtualenv =========================================
True

== tensorflow import ============================================
tf.VERSION = 1.3.0
tf.GIT_VERSION = v1.3.0-rc2-20-g0787eee
tf.COMPILER_VERSION = v1.3.0-rc2-20-g0787eee
Sanity check: array([1], dtype=int32)

== env ==========================================================
LD_LIBRARY_PATH is unset
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
tf_env_collect.sh: line 105: nvidia-smi: command not found

== cuda libs  ===================================================

**Comment by jart** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by adrinjalali** [comment]
Asked on [stackoverflow](https://stackoverflow.com/questions/45975463/tensorflow-example-building-input-functions-with-tf-estimator-doesnt-deliver). No answer there.
